soc/arm/st_stm32/stm32h7/soc_m7.c

```diff
@@ -62,12 +62,12 @@ static int stm32h7_init(const struct device *arg)
 
 	SCB_EnableICache();
 
-#ifndef CONFIG_NOCACHE_MEMORY
-	if (!(SCB->CCR & SCB_CCR_DC_Msk)) {
-		SCB_EnableDCache();
+	if (!IS_ENABLED(CONFIG_DCACHE_DISABLE)) {
```

Comment:
`IS_ENABLE(*_DISABLE)` look clunky. Maybe `DCACHE_DISABLE` should be renamed, because it is strange to check if something is enabled to know that it is disabled.

Reply:
I think that `CONFIG_DCACHE_DISABLE` naming is correct here. If it is set, then the cache shall be **explicitly** disabled . By default `DCache` is enabled most of the time, so introducing `CONFIG_DCACHE_ENABLED` with `default y` seems to me like an overkill. 

The problem is with the current code, where there is `#ifndef CONFIG_NOCACHE_MEMORY`
Maybe below code would be better?

``` 
if (IS_ENABLED(CONFIG_DCACHE_DISABLE)) {
    SCB_DisableDCache();
} else {
    if (!(SCB->CCR & SCB_CCR_DC_Msk)) {
        SCB_EnableDCache();
    }
}
```

However, in this code we do disable already disabled `DCache` when `CONFIG_DCACHE_DISABLED` is defined.